src/Arcus.Messaging.Tests.Integration/MessagePump/ServiceBusMessagePumpTests.cs

```diff
@@ -0,0 +1,118 @@
+    public class ServiceBusMessagePumpTests : IntegrationTest, IAsyncLifetime
+    {
+        private const string QueueConnectionStringKey = "Arcus:ServiceBus:ConnectionStringWithQueue";
+        private const string TopicConnectionStringKey = "Arcus:ServiceBus:ConnectionStringWithTopic";
+
+        public static IEnumerable<object[]> Encodings
+        {
+            get
+            {
+                yield return new object[] { Encoding.UTF8 };
+                yield return new object[] { Encoding.UTF7 };
+                yield return new object[] { Encoding.UTF32 };
+                yield return new object[] { Encoding.ASCII };
+                yield return new object[] { Encoding.Unicode };
+                yield return new object[] { Encoding.BigEndianUnicode };
+            }
+        }
+
+        private ServiceBusEventConsumerHost _serviceBusEventConsumerHost;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="TcpHealthCheckTests" /> class.
+        /// </summary>
+        public ServiceBusMessagePumpTests(ITestOutputHelper testOutput) : base(testOutput)
+        {
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings))]
+        public async Task ServiceBusQueueMessagePump_PublishServiceBusMessage_MessageSuccessfullyProcessed(Encoding messageEncoding)
+        {
+            await ServiceBusMessagePump_PublishServiceBusMessage_MessageSuccessfullyProcessed(messageEncoding, QueueConnectionStringKey);
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings))]
+        public async Task ServiceBusTopicMessagePump_PublishServiceBusMessage_MessageSuccessfullyProcessed(Encoding messageEncoding)
+        {
+            await ServiceBusMessagePump_PublishServiceBusMessage_MessageSuccessfullyProcessed(messageEncoding, TopicConnectionStringKey);
+        }
+
+        private async Task ServiceBusMessagePump_PublishServiceBusMessage_MessageSuccessfullyProcessed(Encoding messageEncoding, string connectionStringKey)
+        {
+            // Arrange
+            var operationId = Guid.NewGuid().ToString();
+            var transactionId = Guid.NewGuid().ToString();
+            var messageSender = CreateServiceBusSender(connectionStringKey);
+
+            var order = OrderGenerator.Generate();
+            var orderMessage = order.WrapInServiceBusMessage(operationId, transactionId, encoding: messageEncoding);
+
+            // Act
+            await messageSender.SendAsync(orderMessage);
+
+            // Assert
+            var receivedEvent = _serviceBusEventConsumerHost.GetReceivedEvent(operationId);
+            Assert.NotEmpty(receivedEvent);
+            var deserializedEventGridMessage = EventGridParser.Parse<OrderCreatedEvent>(receivedEvent);
+            Assert.NotNull(deserializedEventGridMessage);
+            Assert.Single(deserializedEventGridMessage.Events);
+            var orderCreatedEvent = deserializedEventGridMessage.Events.SingleOrDefault();
```

Comment:
`orderCreatedEvent` is actually used before checked. You could also use:
```csharp
var orderCreatedEvent = Assert.Single(deserializedEventGridMessage.Events);
```
Just saying ;)

Reply:
It's used on the line above (83), no? :thinking:

Reply:
Yeah, and again on (84). The `Assert.Single` already returns the single element. Was just a nitpicking improvement. Not blocking.

Reply:
Ah now I get it, good point. Changed it!